packages/teams-js/src/public/tasks.ts

```diff
@@ -27,7 +29,10 @@ export namespace tasks {
     taskInfo: TaskInfo,
     submitHandler?: (err: string, result: string | object) => void,
   ): IAppWindow {
-    return dialog.open(getDialogInfoFromTaskInfo(taskInfo), submitHandler);
+    ensureInitialized(FrameContexts.content, FrameContexts.sidePanel, FrameContexts.meetingStage);
```

Comment:
It might be slightly less error prone to still call dialog.open here and then ignore its return value and just return ChildAppWindow out of this. It would keep the two flows (task.startTask and dialog.open) more in sync which might make it easier for people to convert in the future. Code I'm imagining here would be something like:

dialog.open(getDialogInfoFromTaskInfo(taskInfo), submitHandler ? (res: Result) => submitHandler(result.err, result.result) : undefined);
return new ChildAppWindow();

What do you think?

Reply:
That makes sense. I will make the change.